state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -57,6 +59,16 @@ pub struct Swap {
 	pub fee_taken: bool,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct SwapLegInfo {
+	pub swap_id: SwapId,
+	pub from: Asset,
```

Comment:
Instead of `from` and `to`, we should use `base_asset`, `quote_asset`, and `side: Order` (This type is being changed to be called Side fyi, by Jan).

I'd like to do this for consistency with the new rpcs.